c_src/enif_jq.c

```diff
@@ -311,7 +311,7 @@ static int process_json(
   ERL_NIF_TERM list0 = enif_make_list(env, 0);
   while (jv_is_valid(result = jq_next(jq))) {
       ret = JQ_OK;
-      jv res_jv_str = jv_dump_string(jv_copy(result), dumpopts);
+      jv res_jv_str = jv_dump_string(result, dumpopts);
```

Comment:
I just look though how `jc` use the `jv_dump_string` and I find https://github.com/emqx/jqc/blob/jq-1.6-emqx/src/main.c#L208, Is that a fault as well?

Reply:
Not sure. I will have to look at this a bit more to be sure it is correct. One thing is a bit strange: without this change memory sanitizer report memory leaks but only on some test cases. No memory leaks on for the test cases that were in the eunit test suit before the previous commit ([db2f35f](https://github.com/emqx/jq/pull/17/commits/db2f35f55c8817a6a9bd1db89d45bc9e5ae1cd62)) but memory leaks for almost all test cases added with the previous commit ([db2f35f](https://github.com/emqx/jq/pull/17/commits/db2f35f55c8817a6a9bd1db89d45bc9e5ae1cd62)). I also tried to upgrade the jq dependency to the latest master and then even some of the "original" test cases gave memory leaks. The API is almost without documentation so one have to study the code to figure out how things works.

Reply:
A few lines above the line you point to, there is a loop that is similar to this one, and there the code looks like after this change ([see](https://github.com/emqx/jqc/blob/jq-1.6-emqx/src/main.c#L173)). Not sure if that is correct or not.

Reply:
jv_get_refcnt(result) before the jv_dump_string line sometimes gives 2 and sometimes 1 depending on the input,

Reply:
I will merge this now to make it easier to handle the next PR but I will take a look at this later to try to understand more deeply what is happeing.

Reply:
I have now looked into what was going on. The test cases before this PR did not cause memory leaks because the results were either:

* an intermediate number (an intermediate value is not allocated on the heap, and `jq_copy` and `jq_free` do not have any effect on such values) or,
* an exact copy of the input JSON object in which case [this hacky free loop](https://github.com/kjellwinblad/jq/blob/cc4d1f1f56aa202f157988327401ff63e8bd1ac5/c_src/enif_jq.c#L470) happened to take care of the freeing (this free loop is not necessary anymore as the ref count of the input JSON object is now always 1 after the processing as it should be).